src/dndbeyond/content-scripts/character.js

```diff
@@ -1181,46 +1223,7 @@ function rollSpell(force_display = false, force_to_hit_only = false, force_damag
             damage_types.push(dmgtype);
         }
 
-        handleSpecialSpells(spell_name, damages, damage_types, {spell_level: level, spell_source, castas});
-
-        // Hex blade's curse only applies if there are damages
-        if (damages.length > 0 &&
-            character.getSetting("warlock-hexblade-curse", false) &&
-            character.hasClassFeature("Hexblade’s Curse") &&
-            character._proficiency !== null) {
-            damages.push(character._proficiency);
-            damage_types.push("Hexblade's Curse");
-        }
-        // Protector Aasimar: Radiant Soul Damage
-        if (character.hasRacialTrait("Radiant Soul") &&
-            character.getSetting("protector-aasimar-radiant-soul", false)) {
-            damages.push(character._level);
-            damage_types.push("Radiant Soul");
-        }
-    
-        // Ranger - Gathered Swarm
-        if (character.hasClassFeature("Gathered Swarm") &&
-            character.getSetting("ranger-gathered-swarm", false) &&
-            to_hit !== null) {
-            const ranger_level = character.getClassLevel("Ranger");
-            damages.push(ranger_level < 11 ? "1d6" : "1d8");
-            damage_types.push("Gathered Swarm");
-        }
-        // Ranger - Favored Foe
-        if (character.hasClassFeature("Favored Foe") &&
-            character.getSetting("ranger-favored-foe", false) &&
-            to_hit !== null) {
-            const ranger_level = character.getClassLevel("Ranger");
-            damages.push(ranger_level < 6 ? "1d4" : ( ranger_level < 14 ? "1d6" : "1d8"));
-            damage_types.push("Favored Foe");
-        }
-
-        // Warlock: Genie Patron - Genie's Wrath
-        if (character.hasClassFeature("Genie’s Vessel") && character.getSetting("genies-vessel", false) && to_hit != null) {
-            damages.push(character._proficiency);
-            damage_types.push("Genie's Wrath");
-        }
-    
+        handleSpecialSpells(spell_name, damages, damage_types, {spell_level: level, spell_source, castas}, to_hit);
```

Comment:
You removed the things below here (like hexblade's curse) but you don't call the `handleGenericAttacks` to get them added back, so they are technically gone now.
You might need to add a call to `handleGenericAttacks` but would need to make sure that the things in there would all apply for the use case here (i.e: spells), so that all the conditions are necessarily false apart from the 5 features below that you removed (i.e: sneak attack wouldn't apply because it's not a finesse weapon.. but would it still get triggerred by a "Ranged" spell? because it shouldn't, etc...).

Reply:
Yeah, need to add those things back. It got to the point `handleGenericAttacks` become too bad of a case to handle, so I started to put them into `handleSpecialSpells` instead, but missed those.

Reply:
The only one I forgot was "Hexblade's Curse" 😄